File: VB/TileView-ManualThumbs/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Namespace TileView_ManualThumbs

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private textures As List(Of Texture)

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            InitData()
            gridControl1.DataSource = textures
            tileView1.OptionsTiles.ItemSize = New Size(90, 40)
            AddHandler tileView1.GetThumbnailImage, AddressOf TileView1_GetThumbnailImage
            ' Specify a column that provides information on images to render.
            tileView1.ColumnSet.BackgroundImageColumn = colName
            tileView1.OptionsImageLoad.RandomShow = True
            tileView1.OptionsImageLoad.LoadThumbnailImagesFromDataSource = False
            ' Enable async image load.
            tileView1.OptionsImageLoad.AsyncLoad = True
        End Sub

        Private Sub TileView1_GetThumbnailImage(ByVal sender As Object, ByVal e As DevExpress.Utils.ThumbnailImageEventArgs)
            Dim colorName As String = textures(e.DataSourceIndex).Name
            e.ThumbnailImage = GetImage(e.DesiredThumbnailSize, colorName)
        End Sub

        Private Function GetImage(ByVal imageSize As Size, ByVal colorName As String) As Image
            'Generate a thumbnail
            Dim image As Bitmap = New Bitmap(imageSize.Width, imageSize.Height)
            Using graphics As Graphics = Graphics.FromImage(image)
                Dim tileColor As Color = Color.FromName(colorName)
                Dim grUnit As GraphicsUnit = GraphicsUnit.Pixel
                Dim imageRect As RectangleF = image.GetBounds(grUnit)
                Using brush As LinearGradientBrush = New LinearGradientBrush(imageRect, Color.White, Color.White, 45, False)
                    Dim cblend As ColorBlend = New ColorBlend(4)
                    cblend.Colors = New Color(3) {Color.White, tileColor, tileColor, Color.White}
                    cblend.Positions = New Single(3) {0F, 0.5F, 0.7F, 1F}
                    brush.InterpolationColors = cblend
                    graphics.FillRectangle(brush, imageRect)
                End Using
            End Using

            Return image
        End Function

        Private Sub InitData()
            textures = New List(Of Texture)()
            Dim colorsArray As Array = [Enum].GetNames(GetType(KnownColor))
            For Each colorName In colorsArray
                textures.Add(New Texture(colorName.ToString()))
            Next
        End Sub
    End Class

    Public Class Texture

        Public Sub New(ByVal name As String)
            Me.Name = name
        End Sub

        Public Property Name As String
    End Class
End Namespace

File: CS/TileView-ManualThumbs/Form1.cs
using DevExpress.XtraGrid.Views.Tile;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TileView_ManualThumbs {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        List<Texture> textures;

        private void Form1_Load(object sender, EventArgs e) {
            InitData();

            gridControl1.DataSource = textures;
            tileView1.OptionsTiles.ItemSize = new Size(90, 40);
            tileView1.GetThumbnailImage += TileView1_GetThumbnailImage;
            // Specify a column that provides information on images to render.
            tileView1.ColumnSet.BackgroundImageColumn = colName;
            tileView1.OptionsImageLoad.RandomShow = true;
            tileView1.OptionsImageLoad.LoadThumbnailImagesFromDataSource = false;
            // Enable async image load.
            tileView1.OptionsImageLoad.AsyncLoad = true;
        }

        private void TileView1_GetThumbnailImage(object sender, DevExpress.Utils.ThumbnailImageEventArgs e) {
            string colorName = textures[e.DataSourceIndex].Name;
            e.ThumbnailImage = GetImage(e.DesiredThumbnailSize, colorName);
        }

        private Image GetImage(Size imageSize, string colorName) {
            //Generate a thumbnail
            Bitmap image = new Bitmap(imageSize.Width, imageSize.Height);
            using(Graphics graphics = Graphics.FromImage(image)) {
                Color tileColor = Color.FromName(colorName);
                GraphicsUnit grUnit = GraphicsUnit.Pixel;
                RectangleF imageRect = image.GetBounds(ref grUnit);
                using(LinearGradientBrush brush = new LinearGradientBrush(imageRect, Color.White, Color.White, 45, false)) {
                    ColorBlend cblend = new ColorBlend(4);
                    cblend.Colors = new Color[4] { Color.White, tileColor, tileColor, Color.White };
                    cblend.Positions = new float[4] { 0f, 0.5f, 0.7f, 1f };
                    brush.InterpolationColors = cblend;
                    graphics.FillRectangle(brush, imageRect);
                }
            }
            return image;
        }

        private void InitData() {
            textures = new List<Texture>();
            System.Array colorsArray = Enum.GetNames(typeof(KnownColor));
            foreach(var colorName in colorsArray ) {
                textures.Add(new Texture(colorName.ToString()));
            }
        }
    }

    public class Texture { 
        public Texture(string name) {
            this.Name = name;
        }
        public string Name { get; set; }
    }
}
